nf_core/download.py

```diff
@@ -167,95 +166,187 @@ def download_workflow(self):
+        # Release
         else:
-            log.error("Not able to find pipeline '{}'".format(self.pipeline))
-            log.info("Available pipelines: {}".format(", ".join([w.name for w in wfs.remote_workflows])))
-            raise LookupError("Not able to find pipeline '{}'".format(self.pipeline))
+            for r in self.wf_releases:
+                if r["tag_name"] == self.release:
+                    self.wf_sha = r["tag_sha"]
+                    break
+
+            # Can't find the release or branch - throw an error
+            else:
+                log.info(
+                    "Available {} releases: '{}'".format(
+                        self.pipeline, "', '".join([r["tag_name"] for r in self.wf_releases])
+                    )
+                )
+                log.info("Available {} branches: '{}'".format(self.pipeline, "', '".join(self.wf_branches.keys())))
+                raise AssertionError(
+                    "Not able to find release / branch '{}' for {}".format(self.release, self.pipeline)
+                )
+
+        # Set the outdir
+        if not self.outdir:
+            self.outdir = "{}-{}".format(self.pipeline.replace("/", "-").lower(), self.release)
+
+        # Set the download URL and return
+        self.wf_download_url = "https://github.com/{}/archive/{}.zip".format(self.pipeline, self.wf_sha)
+
+    def prompt_container_download(self):
+        """Prompt whether to download container images or not"""
+
+        if self.container is None:
+            stderr.print("\nIn addition to the pipeline code, this tool can download software containers.")
+            self.container = questionary.select(
+                "Download software container images:",
+                choices=["none", "singularity"],
+                style=nf_core.utils.nfcore_question_style,
+            ).unsafe_ask()
+
+    def prompt_use_singularity_cachedir(self):
+        """Prompt about using $NXF_SINGULARITY_CACHEDIR if not already set"""
+        if (
+            self.container == "singularity"
+            and os.environ.get("NXF_SINGULARITY_CACHEDIR") is None
+            and stderr.is_interactive  # Use rich auto-detection of interactive shells
+        ):
+            stderr.print(
+                "\nNextflow and nf-core can use an environment variable called [blue]$NXF_SINGULARITY_CACHEDIR[/] that is a path to a directory where remote Singularity images are stored. "
+                "This allows downloaded images to be cached in a central location."
+            )
+            if rich.prompt.Confirm.ask(
+                f"[blue bold]?[/] [bold]Define [blue not bold]$NXF_SINGULARITY_CACHEDIR[/] for a shared Singularity image download folder?[/]"
+            ):
+                # Prompt user for a cache directory path
+                cachedir_path = None
+                while cachedir_path is None:
+                    cachedir_path = os.path.abspath(
```

Comment:
```suggestion
                    cachedir_path = os.path.expanduser(
```

tested this locally and that solved the issue for me. the returned path is the absolute path

Reply:
It won't be the absolute path if it doesn't do the expansion, so we still need the `abspath` (since this could possibly go into `.bashrc`).

Reply:
Is it the absolute path when you _don't_ use the `~`? eg you just do a subdir in the pwd called `foo` (not `~/foo` or `/foo`).

I suspect that those relative paths will not be absolute, so we need:

```suggestion
                    cachedir_path = os.path.abspath(os.path.expanduser(
```
(with a matching second `)` below)...

Reply:
Added 👍🏻